### PR TITLE
create account for kvstore

### PR DESCRIPTION
**Pull request by handsomepotato**
feature kvstore support account creation
image

### CODE REVIEW

File: website/docs/r/kvstore_account.html.markdown [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `instance_id` - (Required, ForceNew) The Id of instance in which account belongs. (The engine version of instance must be 4.0 or 4.0+)
+* `name` - (Required, ForceNew) Operation account requiring a uniqueness check. It may consist of lower case letters, numbers, and underlines, and must start with a letter and have no more than 16 characters.
+* `password` - (Optional, Sensitive) Operation password. It may consist of letters, digits, or underlines, with a length of 6 to 32 characters. You have to specify one of `password` and `kms_encrypted_password` fields.
+* `kms_encrypted_password` - (Optional, Available in 1.57.1+) An KMS encrypts password used to a KVStore account. If the `password` is filled in, this field will be ignored.
```
Comment (Lexsss) [inline website/docs/r/kvstore_account.html.markdown]: remove Available

File: website/docs/r/kvstore_account.html.markdown [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+* `kms_encryption_context` - (Optional, MapString, Available in 1.57.1+) An KMS encryption context used to decrypt `kms_encrypted_password` before creating or updating a KVStore account with `kms_encrypted_password`. See [Encryption Context](https://www.alibabacloud.com/help/doc-detail/42975.htm). It is valid when `kms_encrypted_password` is set.
```
Comment (Lexsss) [inline website/docs/r/kvstore_account.html.markdown]: still

File: alicloud/resource_alicloud_kvstore_account_test.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+
+		// module name
+		IDRefreshName: resourceId,
+
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckKVStoreAccountDestroy,
```
Comment (Lexsss) [inline alicloud/resource_alicloud_kvstore_account_test.go]: CheckDestroy: rac.checkResourceDestroy(),

File: alicloud/resource_alicloud_kvstore_account_test.go [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
```
Comment (Lexsss) [inline alicloud/resource_alicloud_kvstore_account_test.go]: CheckDestroy: rac.checkResourceDestroy(),

File: alicloud/resource_alicloud_kvstore_account.go [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+	}
+	request := r_kvstore.CreateDeleteAccountRequest()
+	request.RegionId = client.RegionId
+	request.InstanceId = parts[0]
+	request.AccountName = parts[1]
+
+	raw, err := client.WithRkvClient(func(rkvClient *r_kvstore.Client) (interface{}, error) {
+		return rkvClient.DeleteAccount(request)
+	})
+	if err != nil && !IsExceptedError(err, InvalidAccountNameNotFound) {
```
Comment (Lexsss) [inline alicloud/resource_alicloud_kvstore_account.go]: if error code is InvalidAccountNameNotFound, it should return nil directly without call the wait method.

File: alicloud/resource_alicloud_kvstore_account.go [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+		password := d.Get("password").(string)
+		kmsPassword := d.Get("kms_encrypted_password").(string)
+
+		if password == "" && kmsPassword == "" {
+			return WrapError(Error("One of the 'password' and 'kms_encrypted_password' should be set."))
+		}
+
+		if password != "" {
+			d.SetPartial("password")
```
Comment (Lexsss) [inline alicloud/resource_alicloud_kvstore_account.go]: d.SetPartial("password") move to line 218

File: alicloud/resource_alicloud_kvstore_account.go [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+			request.AccountPassword = password
+		} else {
+			kmsService := KmsService{meta.(*connectivity.AliyunClient)}
+			decryptResp, err := kmsService.Decrypt(kmsPassword, d.Get("kms_encryption_context").(map[string]interface{}))
+			if err != nil {
+				return WrapError(err)
+			}
+			request.AccountPassword = decryptResp.Plaintext
+			d.SetPartial("kms_encrypted_password")
```
Comment (Lexsss) [inline alicloud/resource_alicloud_kvstore_account.go]: these two line still

File: website/docs/r/kvstore_account.html.markdown (hunk 2 above)
Comment (Lexsss) [inline website/docs/r/kvstore_account.html.markdown]: remove MapString

File: website/docs/r/kvstore_account.html.markdown [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+* `description` - (Optional) Database description. It cannot begin with https://. It must start with a Chinese character or English letter. It can include Chinese and English characters, underlines (_), hyphens (-), and numbers. The length may be 2-256 characters.
+* `type` - (Optional, ForceNew)Privilege type of account.
+    - Normal: Common privilege.
```
Comment (Lexsss) [inline website/docs/r/kvstore_account.html.markdown]: - Super:

File: website/docs/r/kvstore_account.html.markdown (hunk 1 above)
Comment (handsomepotato) [reply website/docs/r/kvstore_account.html.markdown]: done

File: website/docs/r/kvstore_account.html.markdown (hunk 2 above)
Comment (handsomepotato) [reply website/docs/r/kvstore_account.html.markdown]: done

File: alicloud/resource_alicloud_kvstore_account_test.go (hunk 3 above)
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account_test.go]: done

File: alicloud/resource_alicloud_kvstore_account_test.go (hunk 4 above)
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account_test.go]: done

File: alicloud/resource_alicloud_kvstore_account.go (hunk 5 above)
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account.go]: done

File: alicloud/resource_alicloud_kvstore_account.go (hunk 7 above)
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account.go]: done

File: website/docs/r/kvstore_account.html.markdown (hunk 2 above)
Comment (handsomepotato) [reply website/docs/r/kvstore_account.html.markdown]: done

File: website/alicloud.erb [hunk 9]
```diff
@@ -959,6 +959,9 @@
                           <li>
                             <a href="/docs/providers/alicloud/r/kvstore_instance.html">alicloud_kvstore_instance</a>
                           </li>
+                          <li>
+                            <a href="/docs/providers/alicloud/r/kvstore_account.html">alicloud_kvstore_account</a>
+                          </li>
```
Comment (xiaozhu36) [inline website/alicloud.erb]: Please sort by alphabet

File: website/docs/r/kvstore_account.html.markdown [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+page_title: "Alicloud: alicloud_kvstore_account"
+sidebar_current: "docs-alicloud-resource-kvstore-account"
+description: |-
+  Provides an kvstore account resource.
+---
+
+# alicloud\_kvstore\_account
+
+Provides an kvstore account resource and used to manage databases.
+
```
Comment (xiaozhu36) [inline website/docs/r/kvstore_account.html.markdown]: Please add a note: Available in 1.66.0+

File: alicloud/resource_alicloud_kvstore_account.go (hunk 6 above)
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account.go]: done

File: alicloud/resource_alicloud_kvstore_account.go [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+		},
+
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"name": {
```
[the remark this answers is not in the document]
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account.go]: done

File: alicloud/resource_alicloud_kvstore_account.go [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+					return d.Get("kms_encrypted_password").(string) == ""
+				},
+				Elem: schema.TypeString,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Normal"}, false),
+				ForceNew:     true,
+				Computed:     true,
```
[the remark this answers is not in the document]
Comment (handsomepotato) [reply alicloud/resource_alicloud_kvstore_account.go]: done

File: website/alicloud.erb (hunk 9 above)
Comment (handsomepotato) [reply website/alicloud.erb]: done

File: website/docs/r/kvstore_account.html.markdown (hunk 10 above)
Comment (handsomepotato) [reply website/docs/r/kvstore_account.html.markdown]: done

**Pull request merged**